protocols/autonat/src/protocol.rs

```diff
@@ -150,28 +154,30 @@ impl DialRequest {
     }
 
     pub fn into_bytes(self) -> Vec<u8> {
+        use quick_protobuf::MessageWrite;
+
         let peer_id = self.peer_id.to_bytes();
         let addrs = self
             .addresses
             .into_iter()
-            .map(|addr| addr.to_vec())
+            .map(|addr| Cow::from(addr.to_vec()))
             .collect();
 
-        let msg = structs_proto::Message {
-            r#type: Some(structs_proto::message::MessageType::Dial as _),
-            dial: Some(structs_proto::message::Dial {
-                peer: Some(structs_proto::message::PeerInfo {
-                    id: Some(peer_id),
+        let msg = proto::Message {
+            type_pb: Some(proto::MessageType::DIAL),
+            dial: Some(proto::Dial {
+                peer: Some(proto::PeerInfo {
+                    id: Some(Cow::from(peer_id)),
                     addrs,
                 }),
             }),
-            dial_response: None,
+            dialResponse: None,
```

Comment:
This is odd. Does `pb-rs` not generate idiomatic Rust identifiers?

Reply:
No it doesn't not :/